pkg/kubelet/prober/worker.go

```diff
@@ -253,7 +253,9 @@ func (w *worker) doProbe(ctx context.Context) (keepGoing bool) {
 		}
 		// Abort if the container will not be restarted.
 		return c.State.Terminated == nil ||
-			w.pod.Spec.RestartPolicy != v1.RestartPolicyNever
+			w.pod.Spec.RestartPolicy != v1.RestartPolicyNever ||
+			(w.container.RestartPolicy != nil && *w.container.RestartPolicy == v1.ContainerRestartPolicyAlways)
```

Comment:
Can we check this only if the container is an init container?
The usage of the container restart policy for a regular container is not defined yet.